examples/ConsPortfolioModel/example_ConsPortfolioModel.py

```diff
@@ -283,8 +302,8 @@ def RiskyShareMertSamLogNormal(RiskPrem, CRRA, RiskyVar):
     agent.solve()
 
     # Compute the analytical Merton-Samuelson limiting portfolio share
-    RiskyVar = agent.RiskyStd ** 2
-    RiskPrem = agent.RiskyAvg - agent.Rfree
+    RiskyVar = agent.RiskyStd**2
+    RiskPrem = agent.RiskyAvg - agent.Rfree[0]
```

Comment:
Is the lack of parallelism between RiskyAvg (a scalar) and RFree (a time-varying something) at all a cause for concern?

Reply:
I think RiskyAvg depends on an input of the portfolio share, and thus is not something whose dynamics are predetermined like those of Rfree?  But if that's right, shouldn't it be Rport?

Reply:
`RiskyAvg` is a model parameter, but not a solver parameter, so it is handled differently than `Rfree`. Depending on it's nature, the agent constructs a `Distribution` that will be set to time varying or time invariant. `Rfree` however is now expected to be time invariant, so even when it's the same throughout the lifecycle, as is here, the agent class holds the value as a list.

What @llorracc is talking about is the method `get_Rfree()` which does make a weighted combination of the risky return (distributed ~ lognormal) and the `Rfree` to return `Rport`. 

In this way, `Rfree` behaves like `PermGroFac` or `LivPrb` which are always lists and are solver parameters. 

Perhaps what you are implying is that `RiskyAvg` should also be handled as time-varying, as are the income distributions.